src/api/commands/itemcategory/ItemCategoryUpdateCommand.ts

```diff
@@ -60,29 +58,40 @@ export class ItemCategoryUpdateCommand extends BaseCommand implements RpcCommand
         } as ItemCategoryUpdateRequest);
     }
 
-    /**
-     * - should have 4 params
-     * - if category has key, it cant be edited
-     * - ...
-     *
-     * @param {RpcRequest} data
-     * @returns {Promise<void>}
-     */
     public async validate(data: RpcRequest): Promise<RpcRequest> {
-
         if (data.params.length < 4) {
-            throw new MessageException('Missing parameters.');
+            throw new MessageException('Missing args. Requires 4 args.');
```

Comment:
add Missing- and InvalidParamExceptions

Reply:
InvalidParamException could be improved by adding optional parameter correctTypes[] allowing you to define the possible types for the parameter.